src/components/Quiz/QuizRadioGroup.tsx

```diff
@@ -21,6 +25,90 @@ import { Question } from "../../types"
 interface CustomRadioProps extends RadioProps {
   index: number
   label: string
+  showAnswer: boolean
+  isSelectedCorrect: boolean
+}
+
+const CustomRadio: React.FC<CustomRadioProps> = ({
+  index,
+  label,
+  showAnswer,
+  isSelectedCorrect,
+  ...radioProps
+}) => {
+  const { state, getInputProps, getRadioProps, htmlProps } =
+    useRadio(radioProps)
+
+  // Memoized values
+  const buttonBg = useMemo<string>(() => {
+    if (!state.isChecked) return "body.inverted"
+    if (!showAnswer) return "primary.base"
+    if (!isSelectedCorrect) return "error.base"
+    return "success.base"
+  }, [state.isChecked, showAnswer, isSelectedCorrect])
+
+  const primaryBaseColor = useToken("colors", "primary.base")
+
+  const focusProps: FlexProps = {
+    outline: showAnswer ? "none" : `1px solid ${primaryBaseColor}`,
+  }
+
+  const controlFocusProps: SquareProps = {
+    bg: showAnswer ? "white" : "primary.pressed",
+  }
+
+  const getRadioControlBg = (): SquareProps["bg"] => {
+    if (showAnswer) return "white"
+
+    if (state.isChecked) return "primary.pressed"
+
+    return "disabled"
+  }
+
+  const getControlLabelColor = (): TextProps["color"] => {
+    if (!showAnswer) return "white"
+
+    if (isSelectedCorrect) return "success.base"
+
+    return "error.base"
+  }
+
+  // Render CustomRadio component
+  return (
+    <chakra.label {...htmlProps} cursor="pointer" data-group w="100%">
```

Comment:
should we remove this one? since now we are also declaring that the child is going to be a group.
```suggestion
    <chakra.label {...htmlProps} cursor="pointer" w="100%">
```

Reply:
Ahh, right!